README.md

```diff
@@ -12,6 +12,51 @@
 - Decorrelates [BRIEF](https://www.cs.ubc.ca/~lowe/525/papers/calonder_eccv10.pdf) features under rotational invariance, leading to better performance in nearest-neighbor applications.
 - Unlike [SURF](https://www.vision.ee.ethz.ch/~surf/eccv06.pdf) and [SIFT(Scale-Invariant Feature Transform)](http://weitz.de/sift/), which are patented algorithms, [ORB](http://www.willowgarage.com/sites/default/files/orb_final.pdf) is free to use.
 
+## Overview
+
+The process of generating matches takes two phases; `finding` and `matching`. `finding`, 
+or identifying interest points in an image, is done using the `findPoints` method. It passes a `cornersArray` to the global `utils` object's `getPoints` object property, which can be stored for later use. `finding` will take a few hundred milliseconds for images close to 1024 by 768 pixels.
+
+Please note that the the "global `utils` object" mentioned above is returned as a parameter to the callback function from where it can be accessed. See this example:
+
+```js
+  new Matcher('path/to/image1.png', 'path/to/image2.png',
+    async function (r) { // r here is the passed utils object
+      res = await r;
+      console.log(res.getPoints); // see output below
+    });
+```
+The output `getPoints` is in the following format:
+```
+[{"x":37,"y":261},
+ {"x":482,"y":402},
+ {"x":84,"y":331}, ...]
+```
+
+`matching` is done with the `findMatchedPoints` function, passes a `matchesArray` to the global `utils` object's `getMatchedPoints` object property with the following format:

```

Comment:
This is a little hard to follow, could you add a code example showing the function in use? Thanks!!!

Reply:
Cool! I've added code links to those functions since they were kind of lengthy. But let me know if you want me to copy them into the docs. Thanks!

Reply:
Just an example of `findMatchedPoints` being run and the parameters for it would be great.